### PR TITLE
indilib: fix udev rules

### PR DESCRIPTION
**Pull request by returntoreality**
This fixes the udev rules so that the indi packages can be added to nixos services.udev.packages option. It also fixes the loading of firmware for devices which need fxload.

## Description of changes

fixes the paths in the provided udev rules so the indi packages can be added to services.udev.packages

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by panicgh — CHANGES_REQUESTED** [review]
There is a missing `--replace` for `/sbin/modprobe` in indilib/lib/udev/rules.d/80-dbk21-camera.rules. It currently looks like this:

```udev
ATTRS{idVendor}=="199e", ATTRS{idProduct}=="8101",  RUN+="/sbin/modprobe -b uvcvideo", RUN+="/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/sh -c 'echo 199e 8101 > /sys/bus/usb/drivers/uvcvideo/new_id'"
ATTRS{idVendor}=="199e", ATTRS{idProduct}=="8207",  RUN+="/sbin/modprobe -b uvcvideo", RUN+="/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/sh -c 'echo 199e 8207 > /sys/bus/usb/drivers/uvcvideo/new_id'"
```

Result of `nixpkgs-review pr 258667` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>indi-full</li>
    <li>indilib</li>
    <li>kstars</li>
    <li>phd2</li>
    <li>stellarium</li>
  </ul>
</details>

**Comment by returntoreality** [comment]
Fixed paths for modprobe as well and made the udev fixups only on Linux, since other platforms do not use udev.

File: pkgs/development/libraries/science/astronomy/indilib/default.nix [hunk 1]
```diff
@@ -58,6 +60,15 @@ stdenv.mkDerivation rec {
   # Socket address collisions between tests
   enableParallelChecking = false;
 
+  postFixup = lib.optionals stdenv.isLinux ''
```
Comment (panicgh) [inline pkgs/development/libraries/science/astronomy/indilib/default.nix:63]:
```suggestion
  postFixup = lib.optionalString stdenv.isLinux ''
```

File: pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix [hunk 2]
```diff
@@ -68,6 +74,19 @@ stdenv.mkDerivation rec {
     "-DWITH_SBIG=off"
   ];
 
+  postFixup = lib.optionals stdenv.isLinux ''
```
Comment (panicgh) [inline pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix:77]:
```suggestion
  postFixup = lib.optionalString stdenv.isLinux ''
```

File: pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix [hunk 3]
```diff
@@ -52,8 +57,17 @@ stdenv.mkDerivation rec {
     done
   '';
 
-  postFixup = ''
-    rm $out/lib/udev/rules.d/99-fli.rules
+  postFixup = lib.optionals stdenv.isLinux ''
```
Comment (panicgh) [inline pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix:60]:
```suggestion
  postFixup = lib.optionalString stdenv.isLinux ''
```

**Review by panicgh — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix (hunk 2 above)
Comment (returntoreality) [reply pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix]: done

File: pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix (hunk 3 above)
Comment (returntoreality) [reply pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix]: done

**Review by sheepforce — APPROVED** [review]

**Comment by panicgh** [comment]
Note that `indilib` now fails on ofborg for aarch64-darwin, but used to compile on it. I'm not sure what caused the issue since most changes are conditional for `stdenv.isLinux`. Maybe the modified `cmakeFlags` are a problem.

**Comment by returntoreality** [comment]
The tests fail (it builds fine) with `C++ exception with description "Bind to /tmp/fakedriver-test: Address already in use" thrown in the test body.` This fails for indilib, where nothing should have changed for darwin. The cmake flags were modified for the firmware derivation, so this should not affect indilib. I have no darwin machine to test, but this seems to me like a spurious failure, not like it was caused by this change.

**Comment by panicgh** [comment]
@ofborg build indilib

**Review by panicgh — APPROVED** [review]
Given that #260455 doesn't build on ofborg either (same error as this PR), but indilib does on Hydra ([1](https://hydra.nixos.org/build/236515659), [2](https://hydra.nixos.org/build/236568667)), I think we should merge this and check later if Hydra reports any issues. It appears to be an ofborg problem on Darwin.

**Pull request merged**